CDCgov/prime-reportstream
# #11221: Create and upload a blob in the send step and support downloading it

This PR starts uploading reports as part of the send step as well as updating the fetch report function to download the report from either send step or the batch step

Test Steps:
Before going through this login into the okta staging and copy your token out

1. Send a report through the legacy pipeline (or run the smoketests on master)
2. Find the id of a report from the send step (it will have the transport result column set)
3. Run
```
curl --location 'http://localhost:7071/api/history/report/{{id}} \
--header 'organization: development' \
--header 'Authorization: Bearer {token}'
```
and verify that it does not download.
4. Switch to this branch
5. Run 
```
curl --location 'http://localhost:7071/api/history/report/{{id}} \
--header 'organization: development' \
--header 'Authorization: Bearer {token}'
```
and make sure it downloads again
6. Find the parent batch report and run
```
curl --location 'http://localhost:7071/api/history/report/{{id}} \
--header 'organization: development' \
--header 'Authorization: Bearer {token}'
```
and verify it downloads.
7. Run the smoketest and find the ReportFile for a sent report.  Verify the bodyUrl field is set and it can be downloaded

## Changes
- Uploads a report to the blob store as part of the send step
- Update the `getReport` function to support downloading a sent or batch report, regardless of whether a file was uploaded during the send step

## Checklist

### Testing

### Process

## Linked Issues
- #11221 

## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
